### ISSUE TITLE
Document "quick start" development procedure for local environment

### ISSUE THREAD

**Issue by mcshanea** [GoogleChromeLabs/wp-sitemaps#6]
As a person wanting to contribute to the project, I'd like to have an easy way to get the project set up locally so I can write code, test PRs, run unit tests and linting, etc.

## Acceptance criteria

- [x] Decide on an approach for local development
- [x] Add documentation for setting up the plugin locally (could specify Chassis or an alternative)

**Comment by joemcgill** [comment]
I've updated the description here to outline what this ticket is trying to achieve. The real question is whether we want to include a local config that matches the development environment we are using while working on the plugin (i.e. a Chassis config) or if we want to only include written instructions that are environment agnostic.

The advantages of including a Chassis config is that we can script a local dev setup that someone can run in one step, e.g., `composer chassis init` rather than having to clone the plugin repo into their environment and run several other commands to get all of the tooling set up. It also allows us to all share a configuration that speeds up development of the team.

However, in practice, many people are going to want to test on their own local setups anyway, so we'll need to create the documentation regardless, and including a Chassis config may come across as being too prescriptive—even if it's what our team prefers at the moment.

**Comment by svandragt** [comment]
Alright  commited code to https://github.com/humanmade/core-sitemaps-quickstart/commits/master to setup a local dev environment. There's a readme file with full instructions which sets up a testing environment for integration and functional testing including a meta test which passes to confirm the testing environment.

@kirstyburgoine We  should discuss the readme as it may overlaps with #10 you might want to pick some of the content.

It doesn't currently install the plugin as that doesn't yet contains a composer.json file, which I can pick up tomorrow.

I also pushed the branch without a commit in master, as a result I had to rename the branch to master as there was no branch to compare against, apologies for this.

**Comment by svandragt** [comment]
https://github.com/humanmade/core-sitemaps-quickstart/pull/1 deals with setting up the plugin within the quickstart environment

**Comment by svandragt** [comment]
![image](https://user-images.githubusercontent.com/594871/66917890-2576cc00-f016-11e9-8c5c-698efe86fbc3.png)

**Comment by joemcgill** [comment]
Hey @svandragt, the setup you have going is looking pretty comprehensive. I'd like  to change directions a little bit, though, and include all of the config for setting up coding standards and unit/behavior testing in the main plugin repo rather than outside it, so that anyone can run sniffs/tests without needing to use our development environment.

I'd recommend having a quick-start process be only responsible for setting up the local Chassis environment and including any additional tooling we're using that we don't expect others to set up when working on the plugin. For example, my local Chassis config includes Query Monitor, Chassis/Tester, Chassis/SequelPro, and Chassis/xdebug in a project/content/config.yaml file with the following:

```
# Domain names to use for Vagrant
#
# The first name will be used as the machine's name, subsequent names will be
# aliased in your hosts file and in nginx
#
# (When overriding, include vagrant.local if you need it)
hosts:
    - xml-sitemaps.local

# Site Configuration
# (When overriding, include all values)
site:
    name: Core XML Sitemaps

# Plugins
#
# Install and activate a list of plugins from wp.org
plugins:
    - query-monitor

# Extensions
#
# Install a list of extensions automatically
extensions:
    - Chassis/Tester
    - Chassis/SequelPro
    - Chassis/xdebug
```

**Comment by joemcgill** [comment]
Additionally, for our purposes, it's really helpful to include xhprof and xhprof_gui for performance profiling. My current suggestions for an extensions list would look like this:

```
# Extensions
#
# Install a list of extensions automatically
extensions:
    - Chassis/Tester
    - Chassis/SequelPro
    - Chassis/xdebug
    - Chassis/xhprof
    - Chassis/chassis_xhgui
```

This will profile each page request and provide helpful information at `http://project-url.local/xhgui/`

**Comment by svandragt** [comment]
Addressed the feedback in #15. Had to disable Chassis/chassis_openssl because chassis/chassis_xhgui is incompatible. Hoping to fix that upstream.
